msal/src/test/java/com/microsoft/identity/client/e2e/utils/AssertUtils.kt

```diff
@@ -0,0 +1,32 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.e2e.utils
+
+import org.junit.Assert
+
+fun assertState(actual: Any, expected: Class<*>) {
+    val condition = expected.isInstance(actual)
+    if (!condition) {
+        Assert.fail("Type comparison failed. Expected: $expected, actual: ${actual.javaClass}")
+    }
```

Comment:
Is there value using this over something like `Assert.assertTrue(state is ExampleState)`? I find the above usages more difficult to follow

Reply:
assertTrue only validates if `state` is of type `ExampleState`. It doesn't, when the assert fails, tell us what the state was (if not `ExampleState`).

Reply:
Ah, I can see how that would be useful 👍